src/request.zig

```diff
@@ -266,7 +266,8 @@ pub const CookieArgs = struct {
 path: ?[]const u8,
 query: ?[]const u8,
 body: ?[]const u8,
-method: ?[]const u8,
+method: ?std.http.Method,
```

Comment:
I propose that the optional makes no sense here. I think an enum (default) value of "invalid" or sth similar makes more sense.

Reply:
fck. does `std.http.Method` support indicating it wasn't possible to parse the method?

Reply:
Hmmm. What about defining our own enum? WDYT? It could even be based on std.http.Method, but add a value for  "invalid".  std.http returns `error.HttpHeadersInvalid` which we can't do so we might as well put an invalid field into the enum.

Reply:
that would make switching more convenient, no need for checking != null

Reply:
I was thinking about a custom enum, but I went with `std.http.Method` simply because it's easier. I agree that .INVALID would be nice, and I will add that.

Reply:
Or should it be called `.Custom` or `.Other`, cuz it's not entirely impossible that someone needs to define a custom HTTP method for whatever they are working on.

Reply:
Maybe even Unexpected? Because we expect a fixed set and did not detect any of those. So it comes unexpected to zap. Or just NonStandard?